### PR TITLE
feat: add basic triage labeling

### PR DESCRIPTION
**Pull request by codebytere**
Addresses a TODO from a [recent Releases meeting](https://docs.google.com/document/d/1XWdD4uAu9m8Gcpiw1j5fLwwZ_hU4rce9JyTpuKU6uM8/edit)

Adds semver labels to several straightforward PR categories, including `docs`, `ci`, `build`, and `test`.

### CODE REVIEW

File: src/add-triage-labels.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      const getSemanticPrefix = (title: string) => title.split(':')[0];
+      
+      // Label Docs PRs as Semver-Patch PRs.
+      const isDocsPR = getSemanticPrefix(pr.title) === SEMANTIC_PREFIXES.DOCS;
+      if (isDocsPR) {
+        await addLabels(context.octokit, {
+          ...context.repo({}),
+          prNumber: pr.number,
+          labels: [
+            SEMVER_LABELS.PATCH,
```
Comment (nornagon) [inline src/add-triage-labels.ts:25]: does this force `semver/patch` for all `docs:` PRs?

If that's the goal, this should probably remove all other `semver/*` labels too.

FWIW I don't agree that docs changes are always `semver/patch`, as many of them do not change the shipped binary.

File: src/add-triage-labels.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            DOCUMENTATION_LABEL
+          ],
+        });
+      }
+
+      const isCIPR = getSemanticPrefix(pr.title) === SEMANTIC_PREFIXES.CI;
+      const isTestPR = getSemanticPrefix(pr.title) === SEMANTIC_PREFIXES.TEST;
+      const isBuildPR = getSemanticPrefix(pr.title) === SEMANTIC_PREFIXES.BUILD;
+
+      // CI, Test, and Build PRs do not affect Semver.
```
Comment (nornagon) [inline src/add-triage-labels.ts:35]: this is _usually_ the case but not _necessarily_. e.g. a `build:` PR could change a GN build flag that affects the output, which would make it `semver/patch`.

Perhaps this should test for an existing `semver/*` label and respect that if it's already present?

Comment (codebytere) [reply src/add-triage-labels.ts:35]: good point - i'll do that!

Comment (nornagon) [reply src/add-triage-labels.ts:35]: probably same with the `docs` one!

File: src/add-triage-labels.ts (hunk 1 above)
Comment (codebytere) [reply src/add-triage-labels.ts:27]: changed it to respect existing semver labels - it now only adds a label if there's none already present.

File: src/add-triage-labels.ts [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    const isBuildPR = getSemanticPrefix(pr.title) === SEMANTIC_PREFIXES.BUILD;
+
+    // CI, Test, and Build PRs do not affect Semver.
+    if (isCIPR || isTestPR || isBuildPR) {
+      await addLabels(context.octokit, {
+        ...context.repo({}),
+        prNumber: pr.number,
+        labels: [SEMVER_NONE_LABEL],
+      });
+    }
```
Comment (nornagon) [inline src/add-triage-labels.ts:45]: nit:

```suggestion
    const semanticPrefix = pr.title.split(':')[0];

    // Label Docs PRs as Semver-Patch PRs.
    const isDocsPR = semanticPrefix === SEMANTIC_PREFIXES.DOCS;
    const isCIPR = semanticPrefix === SEMANTIC_PREFIXES.CI;
    const isTestPR = semanticPrefix === SEMANTIC_PREFIXES.TEST;
    const isBuildPR = semanticPrefix === SEMANTIC_PREFIXES.BUILD;

    if (isDocsPR) {
      await addLabels(context.octokit, {
        ...context.repo({}),
        prNumber: pr.number,
        labels: [SEMVER_LABELS.PATCH, DOCUMENTATION_LABEL],
      });
    } else if (isCIPR || isTestPR || isBuildPR) {
      // CI, Test, and Build PRs do not affect Semver.
      await addLabels(context.octokit, {
        ...context.repo({}),
        prNumber: pr.number,
        labels: [SEMVER_NONE_LABEL],
      });
    }
```

**Review by nornagon — APPROVED** [review]

**Pull request merged**